test/Microsoft.AspNetCore.Mvc.RazorPages.Test/Internal/PageActionInvokerProviderTest.cs

```diff
@@ -299,6 +362,238 @@ public void OnProvidersExecuting_UpdatesEntriesWhenActionDescriptorProviderColle
+        }
+
+        [Fact]
+        public void OnProviderExecuting_ModelPopulates()
+        {
+            // Arrange
+            var type = typeof(TestSetPageWithModel);
+
+            var descriptor = new PageActionDescriptor
+            {
+                RelativePath = "Path1",
+                FilterDescriptors = new FilterDescriptor[0],
+            };
+            var descriptorCollection1 = new ActionDescriptorCollection(new[] { descriptor }, version: 1);
+            var descriptorCollection2 = new ActionDescriptorCollection(new[] { descriptor }, version: 2);
+            var actionDescriptorProvider = new Mock<IActionDescriptorCollectionProvider>();
+            actionDescriptorProvider.SetupSequence(p => p.ActionDescriptors)
+                .Returns(descriptorCollection1)
+                .Returns(descriptorCollection2);
+
+            var loader = new Mock<IPageLoader>();
+            loader.Setup(l => l.Load(It.IsAny<PageActionDescriptor>()))
+                .Returns(CreateCompiledPageActionDescriptor(descriptor, type));
+            
+            Func<PageContext, object> factory = _ => null;
+            Action<PageContext, object> releaser = (_, __) => { };
+            Func<PageContext, object> modelFactory = _ => null;
+            Action<PageContext, object> modelDisposer = (_, __) => { };
+
+            var pageFactoryProvider = new Mock<IPageFactoryProvider>();
+
+            pageFactoryProvider.Setup(f => f.CreatePageFactory(It.IsAny<CompiledPageActionDescriptor>()))
+    .Returns(factory);
+            pageFactoryProvider.Setup(f => f.CreatePageDisposer(It.IsAny<CompiledPageActionDescriptor>()))
+                .Returns(releaser);
+
+            var modelProvider = new Mock<IPageModelFactoryProvider>();
+            modelProvider.Setup(f => f.CreateModelFactory(It.IsAny<CompiledPageActionDescriptor>()))
+                .Returns(modelFactory);
+            modelProvider.Setup(f => f.CreateModelDisposer(It.IsAny<CompiledPageActionDescriptor>()))
+                .Returns(modelDisposer);
+
+            var invokerProvider = CreateInvokerProvider(
+                 loader.Object,
+                 actionDescriptorProvider.Object,
+                 pageFactoryProvider.Object,
+                 modelProvider.Object);
+            var context = new ActionInvokerProviderContext(
+                new ActionContext(new DefaultHttpContext(), new RouteData(), descriptor));
+
+            // Act
+            invokerProvider.OnProvidersExecuting(context);
+
+            // Assert
+            var entry = Assert.Single(invokerProvider.TestCache.Entries);
+
+            Assert.NotNull(entry.Value.ModelFactory);
+            Assert.Equal(2, entry.Value.ActionDescriptor.HandlerMethods.Count);
+        }
+        
+        [Theory]
+        [InlineData(typeof(TestSetPageWithModel))]
+        [InlineData(typeof(TestSetPageModel))]
+        public void PopulateHandlerMethodDescriptors_FindsAll(Type pageType)
```

Comment:
Also, not super sure this is different from https://github.com/aspnet/Mvc/blob/d7bb910114e9232494f6a6751646d86ec3996dd7/test/Microsoft.AspNetCore.Mvc.RazorPages.Test/Internal/PageActionInvokerProviderTest.cs#L76